lib/jsonapi/utils/response/formatters.rb

```diff
@@ -96,9 +106,18 @@ def build_response_document(object, options)
 
           if object.respond_to?(:to_ary)
             records = build_collection(object, options)
-            results.add_result(JSONAPI::ResourcesOperationResult.new(:ok, records, result_options(object, options)))
+
+            if options[:source_resource].present? && options[:relationship_type].present?
+              results.add_result(JSONAPI::RelatedResourcesOperationResult.new(:ok,
+                                                          options[:source_resource],
+                                                          options[:relationship_type],
+                                                          records,
+                                                          result_options(object, options)))
+            else
+              results.add_result(JSONAPI::ResourcesOperationResult.new(:ok, records, result_options(object, options)))
+            end
           else
-            record = turn_into_resource(object, options)
+            record = turn_into_resource(object, options)         
```

Comment:
Hm, got your point, it makes sense.

I was also thinking on the `"links"` member when I made the suggestion above, but it may not be necessary since it seems to be working properly (I'll double check it later).